cmd/nebraska/metrics.go

```
@@ -0,0 +1,114 @@
+func registerNebraskaMetrics() error {
+	err := prometheus.Register(appInstancePerChannelGaugeMetric)
+	if err != nil {
+		return err
+	}
+	err = prometheus.Register(failedUpdatesGaugeMetric)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getMetricsRefreshInterval returns the metrics update Interval key is set in the environment as time.Duration,
+// If not returns the default update interval.
+func getMetricsRefreshInterval() time.Duration {
+	refreshInterval, err := strconv.Atoi(os.Getenv("NEBRASKA_METRICS_UPDATE_INTERVAL"))
+	if err != nil {
+		return defaultMetricsUpdateInterval
+	}
+
+	return time.Second * time.Duration(refreshInterval)
+}
+
+// registerAndInstrumentMetrics registers the application metrics and instruments them in configurable intervals.
+func registerAndInstrumentMetrics(ctl *controller) error {
+	// register application metrics
+	err := registerNebraskaMetrics()
+	if err != nil {
+		return err
+	}
+
+	refreshInterval := getMetricsRefreshInterval()
+
+	metricsTicker := time.NewTicker(refreshInterval)
+
+	go func() {
+		for {
+			<-metricsTicker.C
+			err := calculateMetrics(ctl)
+			if err != nil {
+				logger.Error("Error updating the metrics")
+			}
+		}
+	}()
+
+	return nil
+}
+
+// calculateMetrics calculates the application metrics and updates the respective metric.
+func calculateMetrics(ctl *controller) error {
+	aipcMetrics, err := ctl.api.GetAppInstancesPerChannelMetrics()
+	if err != nil {
+		logger.Error("getMetrics - getting app instances per channel metrics", "error", err.Error())
+		return err
+	}
+
+	for _, metric := range aipcMetrics {
+		appInstancePerChannelGaugeMetric.WithLabelValues(metric.ApplicationName, metric.Version, metric.ChannelName).Set(float64(metric.InstancesCount))
+	}
+
+	fuMetrics, err := ctl.api.GetFailedUpdatesMetrics()
+	if err != nil {
+		logger.Error("getMetrics - getting failed updates metrics", "error", err.Error())
+		return err
```

Same here, could you return a descriptive error here instead?